nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-processors/src/main/java/org/apache/nifi/processors/kafka/KafkaPublisher.java

```diff
@@ -233,12 +230,10 @@ void setProcessLog(ProcessorLog processLog) {
      */
     private void warnOrError(String message, Exception e) {
         if (e == null) {
-            logger.warn(message);
             if (this.processLog != null) {
```

Comment:
This feels a little off to me, @olegz - why not just pass the ComponentLog in to the constructor? It should always be present.